src/main/java/org/eclipse/microprofile/starter/logging/DynamoDBLogger.java

```diff
@@ -0,0 +1,290 @@
+                + "SignedHeaders=" + signedHeaders + ", " + "Signature=" + signature;
+
+    }
+
+    private static Status sendPayload(final String amzDate, final String authorizationHeader, final String dynamoDBJSON) {
+        final HttpURLConnection myURLConnection;
+        try {
+            myURLConnection = (HttpURLConnection) (new URL(endpoint)).openConnection();
+            myURLConnection.setRequestMethod(method);
+            myURLConnection.setRequestProperty("Content-Type", contentType);
+            myURLConnection.setRequestProperty("X-Amz-Date", amzDate);
+            myURLConnection.setRequestProperty("X-Amz-Target", amzTarget);
+            myURLConnection.setRequestProperty("Authorization", authorizationHeader);
+            myURLConnection.setDoOutput(true);
+            try (OutputStream os = myURLConnection.getOutputStream()) {
+                os.write(dynamoDBJSON.getBytes(StandardCharsets.US_ASCII));
+            }
+            Log.log(Level.FINE, myURLConnection.getResponseMessage());
+        } catch (IOException e) {
+            Log.log(Level.SEVERE, "HttpURLConnection failed.", e);
+            return Status.NOK;
+        }
+
+        try {
+            if (myURLConnection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+                if (Log.isLoggable(Level.FINE)) {
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(myURLConnection.getErrorStream()))) {
+                        String line;
+                        while ((line = reader.readLine()) != null) {
+                            Log.log(Level.FINE, line);
+                        }
+                    }
+                }
+                return Status.NOK;
+            }
+        } catch (IOException e) {
+            Log.log(Level.SEVERE, "HttpURLConnection failed while reading error stream.", e);
+            return Status.NOK;
+        }
+
+        if (Log.isLoggable(Level.FINE)) {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(myURLConnection.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Log.log(Level.FINE, line);
+                }
+            } catch (IOException e) {
+                Log.log(Level.SEVERE, "HttpURLConnection failed while reading input stream.", e);
+                return Status.NOK;
+            }
+        }
+
+        return Status.OK;
+    }
+
+    /**
+     * Expensive blocking method that creates signed request and does HTTP POST to AWS DynamoDB API
+     *
+     * @param logData Logged as is except for groupIdArtifacId which is merely used to enhance log message hash.
+     * @return Either Status.OK or Status.NOK if the AWS DynamoDB access is configured.
+     * The caller is not expected to be able to do anything about the error anyway.
+     */
+    public static Status log(final LogData logData) {
```

Comment:
@rdebusscher I decomposed the log method to logical parts and created other methods accordingly.